pkg/cmd/run.go

```diff
@@ -138,6 +138,24 @@ doppler run --mount secrets.json -- cat secrets.json`,
 		// retrieve secrets
 		dopplerSecrets := fetchSecrets(localConfig, enableCache, fallbackOpts, metadataPath, nameTransformer, dynamicSecretsTTL)
 
+		mountPath := cmd.Flag("mount").Value.String()
+		mountFormatString := cmd.Flag("mount-format").Value.String()
+		mountTemplate := cmd.Flag("mount-template").Value.String()
+		maxReads := utils.GetIntFlag(cmd, "mount-max-reads", 32)
+		// only auto-detect the format if it hasn't been explicitly specified
+		shouldAutoDetectFormat := !cmd.Flags().Changed("mount-format")
+		shouldMountFile := mountPath != ""
+		shouldMountTemplate := mountTemplate != ""
+
+		// The potentially dangerous secret names only are only dangerous when they are set
+		// as environment variables since they have the potential to change the default shell behavior.
+		// When mounting the secrets into a file these are not dangerous
+		if !shouldMountFile {
+			err := controllers.CheckForDangerousSecretNames(dopplerSecrets)
+			if err != nil {
```

Comment:
optional but a cool pattern to be aware of:

```suggestion
		        if err := controllers.CheckForDangerousSecretNames(dopplerSecrets); err != nil {
```

Reply:
👍 

I'm undecided which pattern I favor, sometimes I like the concise 1 liner assignment and check, sometimes I prefer the two liner.